test/parse.jl

```diff
@@ -262,3 +262,48 @@ end
 # only allow certain characters after interpolated vars (#25231)
 @test Meta.parse("\"\$x෴  \"",raise=false) == Expr(:error, "interpolated variable \$x ends with invalid character \"෴\"; use \"\$(x)\" instead.")
 @test Base.incomplete_tag(Meta.parse("\"\$foo", raise=false)) == :string
+
+# issue #22480
+@test isnan(parse(Float64, "NaN", 16))
+@test isnan(parse(Float64, "nan", 16))
+@test isnan(parse(Float64, "nAN", 16))
+
+@test parse(Float64, "Inf", 16) === Inf
+@test parse(Float64, "+Inf", 16) === Inf
+@test parse(Float64, "-Inf", 16) === -Inf
+@test parse(Float64, "inf", 16) === Inf
+@test parse(Float64, "inf ", 16) === Inf
+
+@test parse(Float32, "100.1",2) === 4.5f0
+@test parse(Float32, "100.1  ",2) === 4.5f0
+@test parse(Float32, "+100.1",2) === 4.5f0
+@test parse(Float32, "-100.1",2) === -4.5f0
+
+@test parse(Float64, "ff.4", 16) === 255.25
+@test parse(Float64, "+ff.4", 16) === 255.25
+@test parse(Float64, "+ff.4  ", 16) === 255.25
+@test parse(Float64, "-ff.4", 16) === -255.25
+@test parse(Float64, "1.1", 2) === 1.5
+@test parse(Float64, "0.001", 2) === 0.125
+@test parse(Float64, "100",2) === 4.0
+@test parse(Float64, "+100",2) === 4.0
+@test parse(Float64, "100 ",2) === 4.0
+@test parse(Float64, "100\t",2) === 4.0
+@test parse(Float64, "\t100",2) === 4.0
+@test parse(Float64, "-100.0",2) === -4.0
+@test parse(Float64, "0.1", 3) === 0.3333333333333333
+
+@test parse(Float64, "2.3e3",4) === 176.0
+@test parse(Float64, "2.3e-3",4) === 0.04296875
+@test_throws ArgumentError parse(Float64, "2.3e4",4)
+@test parse(Float64, "2.3e10",4) === 704.0
+@test parse(Float64, "2.3e4",5) === 1625.0
+@test parse(Float64, "2.3e23",5) === 3.173828125e9
+@test parse(Float64, "2.3e-23",5) === 2.12992e-9
+@test parse(Float64, "1e3",5) === 125.0
+@test_throws ArgumentError parse(Float64, "1e6",5)
+@test parse(Float64, "1e6",10) === 1000000.0
+
+@test parse(BigFloat, "2.3e-23",5) == big"2.129919999999999999999999999999999999999999999999999999999999999999999999999986e-09"
+
+@test_throws ArgumentError parse(Float64, "14.0", 3)
```

Comment:
A test, where valid digits could be confused with scientific notation would be good.
```
@test parse(Float64, "1e3", 16) == 16^2 + 14*16 + 3
@test parse(Float64, "1e3", 10)  == 1000.
```